clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusStreamingUpdateAgent.java

```diff
@@ -395,11 +406,19 @@ private CacheResult buildCacheResult(StreamingCacheState state,
       Map<String, Job> jobs;
 
       if (state.savedSnapshot) {
+        List<String> missingJobMappings = state.updatedJobs.stream()
+        .filter(j -> !state.jobIdToApp.containsKey(j))
+          .collect(Collectors.toList());
+
+        log.error("{} updatedJobs missing from jobIdToApp cache: {}", getAgentType(), missingJobMappings);
+
         Set<String> changedApplications = state.updatedJobs.stream()
           .map(j -> state.jobIdToApp.get(j))
           .collect(Collectors.toSet());
         changedApplications.remove(null);
 
+        currentApps.addAll(changedApplications);
```

Comment:
This line is the meat of the PR - without there are scenarios where changes may not be applied incrementally between full snapshot cycles.